Source/WebCore/html/HTMLObjectElement.cpp

```diff
@@ -129,25 +129,25 @@ void HTMLObjectElement::parseAttribute(const QualifiedName& name, const AtomStri
     invalidateStyleAndRenderersForSubtree();
 }
 
-static void mapDataParamToSrc(Vector<String>& paramNames, Vector<String>& paramValues)
+static void mapDataParamToSrc(Vector<AtomString>& paramNames, Vector<AtomString>& paramValues)
 {
     // Some plugins don't understand the "data" attribute of the OBJECT tag (i.e. Real and WMP require "src" attribute).
     bool foundSrcParam = false;
-    String dataParamValue;
+    AtomString dataParamValue;
     for (unsigned i = 0; i < paramNames.size(); ++i) {
         if (equalLettersIgnoringASCIICase(paramNames[i], "src"))
             foundSrcParam = true;
         else if (equalLettersIgnoringASCIICase(paramNames[i], "data"))
             dataParamValue = paramValues[i];
     }
     if (!foundSrcParam && !dataParamValue.isNull()) {
-        paramNames.append("src"_s);
+        paramNames.append(AtomString { "src"_s });
```

Comment:
Really want to understand if and why `ConstructFromLiteral` is ever better than this idiom.

Reply:
This is on my TODO list of things to investigate. My bet is that it is no longer useful now that we have ASCIILiteral and we can likely drop it. I'll follow up.